src/main/webapp/show.js

```diff
@@ -0,0 +1,24 @@
+function updateUI(list) {
```

Comment:
This function and `defaultValue` could use a comment explaining what they are doing at a high level.